eng/common/scripts/update-docs-metadata.ps1

```diff
@@ -103,7 +103,16 @@ foreach ($config in $targets) {
     Write-Host ($pkgsFiltered | % { $_.PackageId + " " + $_.PackageVersion })
   
     foreach ($packageInfo in $pkgsFiltered) {
-      $readmeName = "$($packageInfo.PackageId.Replace('azure-','').Replace('Azure.', '').Replace('@azure/', '').ToLower())-readme${suffix}.md"
+      $readmeName = ""
+      if ($NormalizePackageNameFn -and (Test-Path "function:$NormalizePackageNameFn")) 
```

Comment:
@sima-zhu instead of creating this one extra helper function. Can we instead just add this name to the PackageProps? https://github.com/Azure/azure-sdk-for-java/blob/master/eng/common/scripts/Package-Properties.ps1#L4 That way we can still implement it in each language but we don't need to have an independent function that needs to be implemented.

Reply:
The choices here seems to be a common function which grows as we add languages (and carries the potential for future conflicts between languages) or language-specific business logic in an ever expanding `Language-Settings.ps1` (or other collection of modules). 

While the common implementation may be easier to edit quickly it's also easier to make errors that impact other languages and are difficult to detect. For example, in some future hypothetical situation, if we decided to keep the `azure-` prefix for Python packages but continue removing it for C++ packages (which follow a similar format of `azure-core-cpp` on vcpkg) the first implementation would probably break C++ packages and then we would either have to go to language-specific business logic or add the business logic to detect which language is being used in common. This is a hypothetical situation and therefore subject to YAGNI in this instance.

I think we should be consistent with our language-specific business logic and keep it in the language repos.

Reply:
Just to be clear my suggestion is to still keep the logic inside of the language settings (i.e. set it at https://github.com/Azure/azure-sdk-for-net/blob/master/eng/scripts/Language-Settings.ps1#L17) but I suggest creating a property in the PackageProps class such that we can use that common property here.

Reply:
The regex is used for Doc.Ms readme. I don't know if we have other places follow the same regex rule. I can add var to PackageProp, and people can extend the regex if they have other requirements.

Reply:
Double checked the workflow.
We don't have the packagePath and serviceDirectory parameters asking from `Get-dotnet-PackageInfoFromRepo`. Instead, we can put the normalizedPacakgeName inside of https://github.com/Azure/azure-sdk-for-net/blob/c0ee40e16f9e73ce506f53a04dcf7374c08a0448/eng/scripts/Language-Settings.ps1#L53 to achieve the purpose.